### PR TITLE
Restructure controllercontext structs to hold subnet ID correctly

### PR DESCRIPTION
**Pull request by MarcelMue**
Add subnets correctly in test for golden file.

### CODE REVIEW

File: service/controller/clusterapi/v29/resource/tcnp/create.go [hunk 1]
```diff
@@ -310,7 +320,7 @@ func newSubnets(ctx context.Context, cl v1alpha1.Cluster, md v1alpha1.MachineDep
 			},
 			TCCP: template.ParamsMainSubnetsListItemTCCP{
 				Subnet: template.ParamsMainSubnetsListItemTCCPSubnet{
-					Name: key.SanitizeCFResourceName(key.PublicSubnetName(a.AvailabilityZone)),
+					Name: azMap[a.AvailabilityZone],
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/create.go]:
```suggestion
					ID: azMap[a.AvailabilityZone],
```

File: service/controller/clusterapi/v29/detection/detection.go [hunk 2]
```diff
@@ -73,6 +73,7 @@ func (d *Detection) ShouldUpdate(ctx context.Context, cl v1alpha1.Cluster, md v1
 		return false, microerror.Mask(err)
 	}
 
+	// TODO: Need to check if this always works right now as the Spec might now have the subnet IDs which are in the AvailabilityZones struct here.
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/detection/detection.go]: TODO 😛

File: service/controller/clusterapi/v29/resource/tcnp/create.go (hunk 1 above)
Comment (MarcelMue) [reply service/controller/clusterapi/v29/resource/tcnp/create.go]: Done.

File: service/controller/clusterapi/v29/resource/ipam/machine_deployment_persister.go [hunk 3]
```diff
@@ -45,6 +45,10 @@ func (p *MachineDeploymentPersister) Persist(ctx context.Context, subnet net.IPN
 	}
 
 	{
+		if cr.Annotations == nil {
+			cr.Annotations = map[string]string{}
+		}
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/ipam/machine_deployment_persister.go]: Fixing nil map assignment panics here.

File: service/controller/clusterapi/v29/detection/detection.go (hunk 2 above)
Comment (xh3b4sd) [reply service/controller/clusterapi/v29/detection/detection.go]: Still investigating here.

File: service/controller/clusterapi/v29/detection/detection.go [hunk 4]
```diff
@@ -73,7 +72,7 @@ func (d *Detection) ShouldUpdate(ctx context.Context, cl v1alpha1.Cluster, md v1
 		return false, microerror.Mask(err)
 	}
 
-	if !reflect.DeepEqual(cc.Spec.TenantCluster.TCCP.AvailabilityZones, cc.Status.TenantCluster.TCCP.AvailabilityZones) {
+	if !availabilityZonesEqual(cc.Spec.TenantCluster.TCCP.AvailabilityZones, cc.Status.TenantCluster.TCCP.AvailabilityZones) {
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/detection/detection.go]: Wrote some custom comparison here. Added unit tests.

**Comment by xh3b4sd** [comment]
I just merge this into the feature branch now. We continue with this from there on Monday.

**Pull request merged**